samples/azure-quantum/qae-numerical-integration/QuantumAmplitudeEstimation.ipynb

```diff
@@ -0,0 +1,1148 @@
+    "\n",
+    "We use (and adapt) the MLE calculation methods from the work resulting from [arXiv:1904.10246](https://arxiv.org/pdf/1904.10246.pdf) that can be found on [GitHub](https://github.com/qiskit-community/qiskit-community-tutorials/blob/master/algorithms/SimpleIntegral_AEwoPE.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def calculate_theta(hit_list, number, shots):\n",
+    "    \"\"\"\n",
+    "        calculate optimal theta values\n",
+    "            hit_list: list of count of obserbving \"1\" for qc_list\n",
+    "            numebr_grover_list: list of number of Grover operators        \n",
+    "            shots_list: list of number of shots\n",
+    "\n",
+    "        Return:\n",
+    "            theta_candidates: list of optimal theta\n",
+    "    \"\"\"\n",
+    "\n",
+    "    small = 1.e-15  # small valued parameter to avoid zero division\n",
+    "    confidence_level = 5  # confidence level to determine the search range\n",
+    "\n",
+    "    theta_candidates = []\n",
+    "    range_min = 0.0 + small\n",
+    "    range_max = 1.0 - small\n",
+    "    for igrover in range(len(number)):\n",
+    "\n",
+    "        def loglikelihood(p):\n",
+    "            ret = np.zeros_like(p)\n",
+    "            theta = np.arcsin(np.sqrt(p))\n",
+    "            for n in range(igrover + 1):\n",
+    "                ihit = hit_list[n]\n",
+    "                arg = (2 * number[n] + 1) * theta\n",
+    "                ret = ret + 2 * ihit * np.log(np.abs(np.sin(arg))) + 2 * (\n",
+    "                    shots - ihit) * np.log(np.abs(np.cos(arg)))\n",
+    "            return -ret\n",
+    "\n",
+    "        search_range = (range_min, range_max)\n",
+    "        prob_candidate = optimize.brute(loglikelihood, [search_range])[0]\n",
+    "        theta_candidates.append(np.arcsin(np.sqrt(prob_candidate)))\n",
+    "        perror = cramer_rao_error(igrover, shots, prob_candidate, number)\n",
+    "        range_max = min(prob_candidate+confidence_level * perror, 1.0 - small)\n",
+    "        range_min = max(prob_candidate-confidence_level * perror, 0.0 + small)\n",
+    "    return theta_candidates\n",
+    "\n",
+    "\n",
+    "def cramer_rao_error(M, shots, p0, number):\n",
+    "    \"\"\"\n",
+    "        calculate Cramér-Rao lower bound\n",
+    "            M: upper limit of the sum in Fisher information \n",
+    "            shots_list:  list of number of shots\n",
+    "            p0: the true parameter value to be estimated\n",
+    "            numebr_grover_list: list of number of Grover operators        \n",
+    "\n",
+    "        Return:\n",
+    "            square root of Cramér-Rao lower bound:  lower bound on the standard deviation of unbiased estimators\n",
+    "    \"\"\"\n",
+    "    fisher_info = 0\n",
+    "    for k in range(M + 1):\n",
```

Comment:
A sum over a generator or list comprehension would be a lot easier here, I think, than modifying `fisher_info` in the body of a loop.